.github/workflows/tests.yml

```diff
@@ -0,0 +1,63 @@
+name: Tests
+
+on: [push, pull_request, workflow_dispatch]
+
+permissions:
+  contents: read # to fetch code (actions/checkout)
+
+jobs:
+  mri:
+    name: >-
+      ${{ matrix.os }} ${{ matrix.ruby }}
+
+    runs-on: ${{ matrix.os }}
+    if: |
+      !(   contains(github.event.pull_request.title,  '[ci skip]')
+        || contains(github.event.pull_request.title,  '[skip ci]'))
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-20.04, ubuntu-22.04, macos-12, macos-13, windows-2022 ]
+        ruby: [ 2.4, 2.5, 2.6, 2.7, '3.0', 3.1, 3.2, 3.3, head ]
```

Comment:
@MSP-Greg : I just synced this with the lastest master branch and it is failing on CI.
Can you please check if I messed up something?

The latest master has the latest PRs merged, one clean up PR I've just pushed, and the release of v4.3.4.

And as far as Ruby versions on CI, I do not want to run all these, just the Ruby releases that are being maintained by the Ruby core team: 3.1 and above (given 3.0 is EOL in a few weeks).

Reply:
FYI: this is what I had in mind: #87

Reply:
I just did a proper rebase, along with removing Ruby 2.3 and earlier from Actions CI.

> And as far as Ruby versions on CI, I do not want to run all these, just the Ruby releases that are being maintained by the Ruby core team: 3.1 and above (given 3.0 is EOL in a few weeks).

You should run CI matching the gemspec, which is:

https://github.com/jgarber/redcloth/blob/8b10f41636616562d822d62382f64cbce24969e4/redcloth.gemspec#L16